File: python/deploy_instance.py
from google.cloud import run_v2

client = run_v2.InstancesClient()

project_id = "steren-run"
location = "europe-west9"
image = "python" # Official Python image from DockerHub

operation = client.create_instance(
    request={
        "parent": f"projects/{project_id}/locations/{location}",
        "instance": {
            "containers": [{"image": image}]
        }
    }
)

print("Operation started. Waiting for completion...")
response = operation.result()
print(f"Successfully deployed instance: {response.name}")

File: nodejs/deploy-instance.js
import { InstancesClient } from '@google-cloud/run';

const client = new InstancesClient();

const projectId = 'steren-run';
const location = 'europe-west9';
const image = 'node'; // Official Node.js image from DockerHub

const [operation] = await client.createInstance({
  parent: `projects/${projectId}/locations/${location}`,
  instance: {
    containers: [{ image }],
  },
});

console.log('Operation started. Waiting for completion...');
const [response] = await operation.promise();
console.log(`Successfully deployed instance: ${response.name}`);
